### DISCUSSION TITLE
add types of interlocking

### DISCUSSION THREAD

**Discussion by sixdiamants** [railtoolkit/tikz-trackschematic#10] [Ideas]
Currently there's only a single type of interlocking symbol. 
Here's a taxonomy of interlocking systems.   
Apart from the existing generic interlocking symbol, there's a need to distinguish

- Central interlocking
- Decentral interlocking
- IP decentral interlocking (the infamous Gleisfeldkonzentrator)

I'm unsure as to how these are best represented graphically.

Here's a suggestion to start the discussion. The pyramid suggests a control hierarchy.

![image](https://user-images.githubusercontent.com/34092440/152677867-7f417da0-10d1-47fd-9033-5ccf22513aed.png)

I'm unsure as to how to represent a Gfk with a notion of IP-networking.

**Comment by kaat0** [comment]
This seems legit!
What do you believe is the main difference between a centralised and a decentralised interlocking? You probably have a much better insight! 

I have a counter-proposal, based on the objects in the current interlocking symbol:
![interlocking](https://user-images.githubusercontent.com/142348/152702305-0cf18b31-a313-4750-a8eb-4a60e865c63c.png)


What do you think?
But we need better descriptions for the individual states of the interlockings to address them in the attributes.

**Comment by sixdiamants** [comment]
The central interlocking controls a number of sub-interlockings; typical architecture for a larger yard. 

I suspected that the circle and rectangle were a top-view of a sombrero-wearing signalman in front of a control panel. This would match the concept of control centre. 
I prefer separating interlocking from traffic control; the latter is mostly remote these days. 

I would like to see a symbol-hierarchy to represent
- traffic management (at the top of the control pyramid. Sombrero-wearer staff in front of a big panel)
- centralised controller ("top tier IL")
- decentralised contoller ("next tier IL")
- field element controller ("bottom tier control")

here's a proposal but you may have better ideas to represent this notion of hierarchic control

![image](https://user-images.githubusercontent.com/34092440/152854439-d736e626-0ca7-456d-9bc6-2b340bd45dc0.png)

**Comment by kaat0** [comment]
What you call a "control center" is an old mechanical interlocking, of which there are still many in use in England and Germany, for example. I am aware that this is very different in the Netherlands. ;)
This interlocking symbol is needed for old interlockings, because the track vacancy information was/is taken via the window of the interlocking. This is why the location of the interlocking in a track diagram is important. This is also why it should not be confused with a control center.

Why is it important to show the location of the central interlocking, decentralized interlocking, or field element controller in a track schematic?

**Comment by sixdiamants** [comment]
OK for retaining the circle+rectangle symbol for old-school mechanical interlocking. 

Symbols for central IL, decentralised IL and field element controller allow planners to assess the need for cabling and access for maintenance - quite a different use case. Can you suggest any symbols for depicting the notion of "hierarchic control" other than the pyramids that I proposed?

The control centre is interesting because it is associated with the concept of control area, One often outlines the control areas and their borders for operational reasons. 

Howzat? The triangles would be would be unstaffed equipment housings, 
![image](https://user-images.githubusercontent.com/34092440/153209009-8b1ed070-5ff8-45cc-9f99-a61fab8de7f3.png)

**Comment by kaat0** [comment]
I'm not entirely clear on the difference between a centralized and decentralized interlocking. perhaps "centralized" is the wrong attribute to describe its meaning.

in general, a three-level pyramid is a good way to represent a hierarchy. Since the library aims to show track schematics and not interlocking cabling, i am somewhat ambivalent about how far the specific needs of cabling should be covered. for the operational focus, however, it makes sense to be able to show how operations are controlled.

I would like to use the rectangle for the building like for the manual interlocking also for the other interlocking types to be able to group the symbols visually,

I have played around a bit to see if the symbols can complement each other well:
![interlocking2](https://user-images.githubusercontent.com/142348/153463566-187d1757-ebe3-4176-bdad-8ccbd3bb1ce0.png)
